activerecord/lib/active_record/validations/uniqueness.rb

```diff
@@ -29,7 +29,22 @@ def validate_each(record, attribute, value)
           end
         end
         relation = scope_relation(record, relation)
-        relation = relation.merge(options[:conditions]) if options[:conditions]
+
+        if options[:conditions]
+          conditions = options[:conditions]
+
+          relation = if conditions.arity.zero?
+            relation.merge(conditions)
```

Comment:
Since `options[:conditions]` [can only be a callable](https://github.com/rails/rails/blob/4174d04f5bfaa451e150b8f97d6050dcfc51233d/activerecord/lib/active_record/validations/uniqueness.rb#L7), I think we should skip the [indirection via `Relation#merge`](https://github.com/rails/rails/blob/4174d04f5bfaa451e150b8f97d6050dcfc51233d/activerecord/lib/active_record/relation/spawn_methods.rb#L48) and call `instance_exec` here directly, so it's more obvious how this branch differs from the one below:

```suggestion
            relation.instance_exec(&conditions)
```